x-pack/platform/packages/shared/kbn-ai-assistant/src/utils/get_model_options_for_inference_endpoints.ts

```diff
@@ -86,22 +89,25 @@ export const getModelOptionsForInferenceEndpoints = ({
 }: {
   endpoints: InferenceAPIConfigResponse[];
 }): ModelOptionsData[] => {
-  // TODO: add logic to show the EIS models if EIS is enabled, if not show the other models
-  const preConfiguredEndpoints = endpoints
-    .map((endpoint) => {
-      const meta = PRECONFIGURED_INFERENCE_ENDPOINT_METADATA[endpoint.inference_id];
+  const hasElserEIS = endpoints.some((ep) => ep.inference_id === ELSER_IN_EIS_INFERENCE_ID);
 
-      if (!meta) {
-        return undefined;
+  return endpoints
+    .filter((endpoint) => {
+      // if ELSER exists in EIS, skip the other ELSER model
+      if (endpoint.inference_id === '.elser-2-elasticsearch' && hasElserEIS) {
```

Comment:
Can you move these constants to a shared file in common? Also extract `.elser-2-elasticsearch` from

 https://github.com/elastic/kibana/blob/1070ceeea2ba62d8d8e1d1ef33d5c9d60cfa7a94/x-pack/platform/plugins/shared/observability_ai_assistant/server/service/index_assets/update_existing_index_assets.ts#L16